src/AutoRest.CSharp/Mgmt/Generation/ResourceWriter.cs

```diff
@@ -98,8 +98,11 @@ protected virtual void WriteCtors()
             _writer.WriteXmlDocumentationSummary($"Initializes a new instance of the <see cref = \"{TypeOfThis.Name}\"/> class.");
             _writer.WriteXmlDocumentationParameter("options", $"The client parameters to use in these operations.");
             _writer.WriteXmlDocumentationParameter("resource", $"The resource that is the target of operations.");
+
+            var idPropString = _resourceData.IsIdString() ? "new ResourceIdentifier(resource.Id)" : "resource.Id";
```

Comment:
Is it guaranteed that parameter always called `resource`?

Reply:
yes